clippy_lints/src/methods/mod.rs

```diff
@@ -3832,6 +3869,7 @@ impl Methods {
                     if let Some(("collect", _, _, span, _)) = method_call(recv) {
                         unnecessary_join::check(cx, expr, recv, join_arg, span);
                     }
+                    else {join_absolute_path::check(cx, expr,  join_arg, span);}
```

Comment:
The `expr` value is the whole `<receiver>.join(<arg>)` expression. For the lint we want to check the type of the receiver. You can simply update this to:

```suggestion
                    else {
                        join_absolute_path::check(cx, recv, join_arg, span)
                    }
```